okio/src/main/java/okio/Buffer.java

```diff
@@ -529,6 +529,41 @@ public byte getByte(long pos) {
     return new ByteString(readByteArray(byteCount));
   }
 
+  @Override public int select(List<ByteString> byteStrings) {
+    Segment s = head;
+    if (s == null) return byteStrings.indexOf(ByteString.EMPTY);
```

Comment:
Interesting behavior. Haven't made it to the tests yet, but I would have just expected -1 here.